packages/components/src/illustrated-message/src/IllustratedMessage.tsx

```diff
@@ -0,0 +1,77 @@
+import { Box } from "../../box";
+import { ComponentProps, ReactNode, forwardRef, useMemo } from "react";
+import { Flex } from "../../layout";
+import { InternalProps, OmitInternalProps, SlotProps, StyledComponentProps, cssModule, mergeProps, useSlots } from "../../shared";
+import { ResponsiveProp, useResponsiveValue } from "../../styling";
+import { Text } from "../../typography";
+
+const DefaultElement = "div";
+
+export interface InnerIllustratedMessageProps extends SlotProps, InternalProps, StyledComponentProps<typeof DefaultElement> {
+    /**
+     * React children.
+     */
+    children: ReactNode;
+    /**
+     * The orientation of the illustrated message.
+     */
+    orientation?: ResponsiveProp<"horizontal" | "vertical">;
+}
+
+export function InnerIllustratedMessage({
+    as = DefaultElement,
+    children,
+    forwardedRef,
+    orientation = "vertical",
+    ...rest
+}: InnerIllustratedMessageProps) {
+    const orientationValue = useResponsiveValue(orientation);
+
+    const { content, heading, image } = useSlots(children, useMemo(() => ({
+        _: {
+            required: ["image"]
```

Comment:
It doesn't really make sense anymore to use an IllustratedMessage component without an header and a content.

Therefore, all slots should be required, meaning: `required: ["image", "content", "heading"]`